### PR TITLE
Added more type information (v2)

### PR DESCRIPTION
**Pull request by ssbarnea**
Added more type information, covering formatters and skip_utils.

Partial: #893

### CODE REVIEW

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

File: lib/ansiblelint/formatters/__init__.py [hunk 1]
```diff
@@ -41,69 +43,72 @@ def _format_path(self, path):
...
-                                    colorize(match.message, Color.error_title),
-                                    colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber),
-                                    colorize(u"{0}".format(match.details), Color.line))
+            return formatstr.format(
+                colorize(u"[{0}]".format(_id), Color.error_code),
+                colorize(match.message, Color.error_title),
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber),
+                colorize(u"{0}".format(match.details), Color.line))
```
Comment (webknjaz) [inline lib/ansiblelint/formatters/__init__.py:55]: looks unrelated

File: lib/ansiblelint/formatters/__init__.py [hunk 2]
```diff
@@ -41,69 +43,72 @@ def _format_path(self, path):
...
+    def format(self, match: MatchError, colored: bool = False) -> str:
         formatstr = u"{0} {1}:{2}"
         if colored:
-            return formatstr.format(colorize(u"[{0}]".format(match.rule.id), Color.error_code),
-                                    colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber))
+            return formatstr.format(
+                colorize(u"[{0}]".format(match.rule.id), Color.error_code),
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber))
```
Comment (webknjaz) [inline lib/ansiblelint/formatters/__init__.py:72]: unrelated?

File: lib/ansiblelint/formatters/__init__.py [hunk 3]
```diff
@@ -41,69 +43,72 @@ def _format_path(self, path):
...
         if colored:
-            return formatstr.format(colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber),
-                                    colorize(u"E{0}".format(match.rule.id), Color.error_code),
-                                    colorize(u"{0}".format(match.message), Color.error_title))
+            return formatstr.format(
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber),
+                colorize(u"E{0}".format(match.rule.id), Color.error_code),
+                colorize(u"{0}".format(match.message), Color.error_title))
```
Comment (webknjaz) [inline lib/ansiblelint/formatters/__init__.py:87]: unrelated?

File: lib/ansiblelint/formatters/__init__.py (hunk 1 above)
Comment (ssbarnea) [reply lib/ansiblelint/formatters/__init__.py:55]: **** it is related! If you would try to add some typing information you will see it. Typing complains about not allowing None, so you add `or ""` part and the line becomes too long, so you need to wrap it to pass the linters job.

Also, colorize requires color Literals, not strings but we forgot to update some calls when we did it. Now mypy does slap us, forcing to fix the bug.

You will see more of these. That is why I said is better to add typing info in chunks that keep the review size decent (ideally under 50 lines).

File: lib/ansiblelint/formatters/__init__.py (hunk 2 above)
Comment (ssbarnea) [reply lib/ansiblelint/formatters/__init__.py:72]: ditto, not trying to slip anything not required.

File: lib/ansiblelint/formatters/__init__.py (hunk 3 above)
Comment (ssbarnea) [reply lib/ansiblelint/formatters/__init__.py:87]: ditto, not trying to slip anything not required.

File: lib/ansiblelint/formatters/__init__.py (hunk 2 above)
Comment (webknjaz) [reply lib/ansiblelint/formatters/__init__.py:72]: I guess all this formatting change makes it look like a formatting change. You shouldn't force reviewers to force to compare the diff byte-by-byte hoping to find something legit in the middle for the reformatting.
